src/middlewared/middlewared/service.py

```diff
@@ -206,6 +151,35 @@ def config(self):
         return await self.do_update(data)
 
 
+class SystemServiceService(ConfigService):
+    """
+    Service service abstract class
+
+    Meant for services that manage system services configuration.
+    """
```

Comment:
This service will be registered as a public service and available to users. Perhaps we need a way to identify classes as abstract?

e.g.
```
class Config:
    abstract = True
```
?

Reply:
Then all the child classes will need to redefine `abstract = False` to become public

We can solve this problem with decorator or metaclass approach, but I am fine with current solution: https://github.com/freenas/freenas/pull/328/commits/3ff019a5e540dd888568d803286bc93a43f0bd37

Reply:
Unless the default was `False` ;)
Config is not inherited.

But yeah, 3ff019a is fine.